source/particle/world.cc

```diff
@@ -263,9 +264,39 @@ namespace aspect
               {
                 const unsigned int n_particles_in_cell = particle_handler->n_particles_in_cell(cell);
 
+                /*
+                 * If using the uniform box generator with add/remove particles and cells without
+                 * particles are allowed, check whether any vertex is outside of the specified box.
+                 * If a vertex is outside, no particles will be generated in that cell, and any
+                 * particles that enter the cell will be removed.
+                 */
+                bool cell_in_domain = true;
+                if (dynamic_cast<const Generator::UniformBox<dim> *>(generator.get()) != nullptr)
```

Comment:
We try to not use `dynamic_cast` any more. You can instead do the following:
```suggestion
                if (plugins::plugin_type_matches<const Generator::UniformBox<dim>>(generator.get()))
```